robot_framework/process.py

```diff
@@ -1,8 +1,142 @@
     orchestrator_connection.log_trace("Running process.")
+    process_arguments = json.loads(orchestrator_connection.process_arguments)
+    service_cvr, certificate_path, data_recipient = (process_arguments[arg] for arg in process_arguments)
+
+    # Prepare access to service platform
+    kombit_access = KombitAccess(service_cvr, certificate_path, True)
+
+    # Receive queue item list of people who weren't registered last time
+    queue_elements = orchestrator_connection.get_queue_elements(config.ROBOT_NAME)
+
+    # Find current list of people with unknown address and get their registration status for Digital Post
+    current_status = get_registration_status_from_query(kombit_access, config.DATABASE, config.QUERY)
+    changes = []
+
+    # Check status of people who are registered against people who weren't registered before
+    for row in queue_elements:
+        data = json.loads(row.data.replace("True", "true").replace("False", "false"))
+        current_value = current_status.get(row.reference, None)
+        if current_value:
+            # If a change has occured since last run, update queue element and prepare to send line to case worker
+            if current_value["digital_post"] != data["digital_post"] or current_value["nemsms"] != data["nemsms"]:
+                orchestrator_connection.delete_queue_element(row.id)
+                orchestrator_connection.create_queue_element(config.ROBOT_NAME, reference=row.reference, data=f'{{"digital_post":{current_value["digital_post"]}, "nemsms":{current_value["nemsms"]}}}')
+                changes.append([current_value["cpr"], current_value["digital_post"], current_value["nemsms"]])
+            current_status.pop(row.reference)
+        else:
+            orchestrator_connection.delete_queue_element(row.id)
+
+    # Send an email with list of people whose status has changed
+    if len(changes) > 0:
+        return_sheet = write_data_to_output_excel(changes)
+        _send_status_email(data_recipient, return_sheet)
+
+    # Add queue elements for elements that didn't exists before
+    for key, current_value in current_status.items():
+        orchestrator_connection.create_queue_element(config.ROBOT_NAME, reference=key, data=f'{{"digital_post":{current_value["digital_post"]}, "nemsms":{current_value["nemsms"]}}}')
+
+
+def write_data_to_output_excel(data: List[Tuple[str, bool, bool]]) -> BytesIO:
+    """ Add data to excel sheet.
+
+    Args:
+        data: A list of tuples with cpr, status for digital post and nem sms
+        target_sheet: A sheet with id's in the first row
+    """
+    wb = Workbook()
+    ws = wb.active
+    ws.append(["CPR", "Digital Post", "NemSMS"])
+
+    for row in data:
+        ws.append(row)
+
+    excel_buffer = BytesIO()
+    wb.save(excel_buffer)
+    excel_buffer.seek(0)
+
+    return excel_buffer
+
+
+def encrypt_data(data, salt) -> str:
+    """Take CPR and first name from citizen and encrypt using the name as salt.
```

Comment:
When the comment asks directly for cpr and name, why not call the args cpr and name?